src/pc/1.13/ChunkColumn.js

```diff
@@ -199,6 +199,10 @@ module.exports = (Block, mcData) => {
         smartBuffer.writeInt32BE(biome)
       })
 
+      if (!smartBuffer.length) {
+        return Buffer.alloc(4096)
```

Comment:
What are these zeros representing? Do you have a link to minecraft code to explain?

Reply:
It's just a workoaround so the load method (which expects the buffer to be not empty) doesn't crash. I'm not sure why I used the size of 4096. I see the native sever uses the length of 1024.

Reply:
There should not be any "workarounds" in stable API methods. The libs should adhere to whatever the vanilla behavior is. In this case "load" deserializes a vanilla chunk from the network and "dump" serializes something to be sent over the network.

Reply:
Yes, in this case dump works incorrectly then. Afaik even minecraft-protocol doesn't allow serialization of empty buffers. I understand there should be no workarounds, can you elaborate on the correct way to fix it?

Reply:
Allocating a buffer of the right size

Reply:
Ok! Will also try to add a test

Reply:
>Allocating a buffer of the right size

@rom1504 ok, got it fixed. should be good now

Reply:
> It's just a workoaround so the load method (which expects the buffer to be not empty) doesn't crash

which load method do you mean? if you mean vanilla one; then ok we should do that and leave a comment to explain this is to address a bug in the vanilla implementation
if it's our implementation of load, then we should fix the bug

Reply:
> and checked that native server always sends the buffer filled with 0 in these cases

this seems to suggest the mojang server; is that indeed what you mean? Do you know why they have this bug?

Reply:
I don't think it's a bug, most probably the vanilla server just tried to normalize the data so the client marks received chunk as empty (otherwise it would print waiting for chunks)

Reply:
And I was about our load method. Not sure what should be fixed in that method instead